### ISSUE TITLE
sentry-cli : no such file or directory while building in Xcode

### ISSUE THREAD

**Issue by akshitrcrm** [getsentry/sentry-react-native#2067]
**OS:**
- [ ]  Windows
- [*]  MacOS
- [ ]  Linux

**Platform:**
- [*]  iOS
- [ ]  Android

**SDK:**
- [*]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 3.2.13

**`react-native` version:** 0.67.2

**Are you using Expo?**
- [ ]  Yes
- [*]  No

**Are you using sentry.io or on-premise?**
- [ ]  sentry.io (SaaS)
- [*]  on-premise

**Configuration:**

(`@sentry/react-native`)
```
Sentry.init({
  dsn: 'https://...@sentry.io/...'
  // other options
});
```
---
**I have following issue:**

Xcode's "Bundle React native code and images" not able to find sentry-cli file.
Give PhaseScriptExecution error:
env: node:  no such file or directory

**Steps to reproduce:**
- brew install node
- install sentry in react native project
- run sentry wizard
- build project in Xcode

**Actual result:**

PhaseScriptExecution error:
env: node:  no such file or directory

**Expected result:**

Build to be succeeded.

I am not using nvm. There is no nvm in my machine. 
when I run nvm command it says command not found

when I run which node it gives 
/opt/homebrew/bin/node

**Comment by akshitrcrm** [comment]
<img width="836" alt="Screenshot 2022-02-17 at 12 10 31 AM" src="https://user-images.githubusercontent.com/69840129/154333677-510f9057-b987-4cab-9c98-34b3dda5cd4d.png">

**Comment by akshitrcrm** [comment]
<img width="567" alt="Screenshot 2022-02-17 at 12 11 31 AM" src="https://user-images.githubusercontent.com/69840129/154333820-d38a044b-2449-4f6e-8825-a6c48358794d.png">

When I run that script in terminal it gives the actual error

**Comment by akshitrcrm** [comment]
<img width="575" alt="damn" src="https://user-images.githubusercontent.com/69840129/154334570-a611d249-5e2c-4da1-b8b9-a733468f8cf6.png">
<img width="811" alt="damn2" src="https://user-images.githubusercontent.com/69840129/154334770-4ca47795-c081-4f81-a433-8377a54d035a.png">

**Comment by akshitrcrm** [comment]
<img width="575" alt="damn" src="https://user-images.githubusercontent.com/69840129/154335046-0359be78-533e-4972-a36c-017bd3172ce2.png">
<img width="811" alt="damn2" src="https://user-images.githubusercontent.com/69840129/154335055-fdb92c98-7589-41b9-9fd2-cda8f31804d7.png">

**Comment by marandaneto** [comment]
@akshitrcrm That works for me on a fresh project, likely something with your environment?
Try to install it standalone? https://docs.sentry.io/product/cli/installation/#installation-via-npm

**Comment by akshitrcrm** [comment]
yeah. I also checked by creating a fresh project on Mac m1, it works, but my old project with sentry doesn't.
I even upgraded my old project to latest react native. still no luck.

**Comment by marandaneto** [comment]
It should be a macOS m1 then, I just ran again on a normal macOS and it works.

@kamilogorek or @AbhiPrasad  are you aware of any issues on Sentry Wizard or Sentry CLI (latest versions) on macOS m1?

**Comment by kamilogorek** [comment]
No, we have arm builds for M1. The issue here appears to be npm configured to not run post-install scripts, which then skips fetching of cli. You can confirm that by going into `node_modules/@sentry/cli` and running `./scripts/install.js`.

**Comment by akshitrcrm** [comment]
> ./scripts/install.js
<img width="744" alt="Screenshot 2022-02-21 at 5 23 48 PM" src="https://user-images.githubusercontent.com/69840129/154950415-61700a83-c6ee-4726-8000-eea54c9105fb.png">

hm it says permission denied...

**Comment by kamilogorek** [comment]
Does xcode still breaks if the binary is there in place now? If so, maybe generated paths are pointing incorrectly?

**Comment by akshitrcrm** [comment]
> Does xcode still breaks if the binary is there in place now? If so, maybe generated paths are pointing incorrectly?

yep, this is what I have in the script
`export SENTRY_PROPERTIES=sentry.properties
export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"
export NODE_BINARY=node
../node_modules/@sentry/cli/bin/sentry-cli react-native xcode   ../node_modules/react-native/scripts/react-native-xcode.sh
`
<img width="694" alt="Screenshot 2022-02-21 at 6 39 37 PM" src="https://user-images.githubuserconte
<img width="875" alt="Screenshot 2022-02-21 at 6 41 45 PM" src="https://user-images.githubusercontent.com/69840129/154962295-c27d31a2-1ba2-4c27-85f8-a196b5666349.png">
nt.com/69840129/154962270-c0d7043c-1d24-4bde-8e44-c30f1e98ab2a.png">

**Comment by akshitrcrm** [comment]
I ran the pwd command and it gives me the correct path, which is inside `ios` folder. So the path is correct as we're moving out of it and accessing `node_modules` directory.

I installed `sentry-cli `globally but I could not access it through terminal for some reason. Maybe I missed adding the path to `zshrc`?
Then I installed sentry-cli from `homebrew` using the command I found in installation docs.
now I can execute sentry-cli on the terminal, it works but Xcode can't. 
I changed the ` ../node_modules/@sentry/cli/bin/sentry-cli` to  `sentry-cli`
but still it is giving me error:
`/Script-00DD1BFF1BD5951E006B06BC.sh: line 6: sentry-cli: command not found`

**Comment by kamilogorek** [comment]
The only possible issue now then is that you are running `bash` or `zsh` interpreter inside your terminal, where XCode is running scripts though `sh` itself. In this case, you need to either:
- move your `PATH` overrides from `.bashrc` (or `.zshrc` if using `zsh`) to `.profile`
- update Xcode shell to use `/bin/bash`, `/usr/local/bin/zsh` or whatever interpreter you are using

**Comment by grzegorzkrukowski** [comment]
I am getting the same response even when running sentry-cli directly not through Xcode.
We had working CI setup for a long time, but now it stopped working (2 days ago).

**Comment by mertozylmz** [comment]
Hi guys,
I solved this problem finally.
All you have to do is write this code in the terminal

```
$ which node
(example export -> `/opt/homebrew/Cellar/node@14/14.17.6/bin/node`)
```
Then write this
`$ ln -s /opt/homebrew/Cellar/node@14/14.17.6/bin/node /usr/local/bin/node`

it worked after i did it

**Comment by gonchar** [comment]
I've found a temp solution which works for me and well explains why is that happening in this article https://mgrebenets.github.io/xcode/2019/04/04/xcode-build-phases-and-environment
I used `/bin/sh -l`
becuase in my case Xcode path doesn't contain even /usr/local/bin

**Comment by kakulgupta** [comment]
Faced the same issue when trying to set up sentry with nvm. 
Ended up adding the following workaround to the build phase scripts 

```sh
REACT_NATIVE_DIR="$(cd "$LOCROOT/../node_modules/react-native" && pwd)"
source "$REACT_NATIVE_DIR/scripts/find-node.sh"
...
...
export NODE_BINARY=node
```
